spec/rubocop/cop/rails/timecop_spec.rb

```diff
@@ -0,0 +1,177 @@
+            .to(eq('freeze_time { }'))
+        end
+      end
+
+      context 'with arguments' do
+        it 'adds an offense' do
+          expect_offense(<<-RUBY.strip_indent)
+            Timecop.freeze(123) { }
+            ^^^^^^^^^^^^^^^^^^^ Use `travel` or `travel_to` instead of `Timecop.freeze`
+          RUBY
+        end
+
+        it 'does not autocorrect' do
+          source = 'Timecop.freeze(123) { }'
+
+          expect(autocorrect_source(source)).to(eq(source))
+        end
+      end
+    end
+  end
+
+  describe 'Timecop.return' do
+    context 'without a block' do
+      it 'adds an offense' do
+        expect_offense(<<-RUBY.strip_indent)
+          Timecop.return
+          ^^^^^^^^^^^^^^ Use `travel_back` instead of `Timecop.return`
+        RUBY
+      end
+
+      context 'in Rails < 6.0', :rails5 do
+        it 'autocorrects to `travel_back`' do
+          expect(autocorrect_source('Timecop.return')).to(eq('travel_back'))
+        end
+
+        context 'inside a block' do
+          it 'autocorrects to `travel_back`' do
+            expect(autocorrect_source('foo { Timecop.return }'))
+              .to(eq('foo { travel_back }'))
+          end
+        end
+      end
+
+      # context 'in Rails > 6.0', :rails6 do
+      #   it 'autocorrects to `unfreeze`' do
+      #     expect(autocorrect_source('Timecop.return')).to(eq('unfreeze'))
+      #   end
+
+      #   context 'inside a block' do
+      #     it 'autocorrects to `unfreeze`' do
+      #       expect(autocorrect_source('foo { Timecop.return }'))
+      #         .to(eq('foo { unfreeze }'))
+      #     end
+      #   end
+      # end
+    end
+
+    context 'with a block' do
+      it 'adds an offense' do
+        expect_offense(<<-RUBY.strip_indent)
+          Timecop.return { }
+          ^^^^^^^^^^^^^^ Use `travel_back` instead of `Timecop.return`
+        RUBY
+      end
+
+      it 'does not autocorrect' do
+        expect(autocorrect_source('Timecop.return { }'))
+          .to(eq('Timecop.return { }'))
+      end
+
+      context 'inside a block' do
+        it 'does not autocorrect' do
+          expect(autocorrect_source('foo { Timecop.return { } }'))
+            .to(eq('foo { Timecop.return { } }'))
+        end
+      end
+    end
+  end
+
+  describe 'Timecop.travel' do
+    it 'adds an offense' do
+      expect_offense(<<-RUBY.strip_indent)
+        Timecop.travel(123) { }
+        ^^^^^^^^^^^^^^^^^^^ Use `travel` or `travel_to` instead of `Timecop.travel`. If you need time to keep flowing, simulate it by travelling again.
+      RUBY
+    end
+  end
+
+  describe 'Timecop.*' do
+    it 'adds an offense' do
+      expect_offense(<<-RUBY.strip_indent)
+        Timecop.foo
+        ^^^^^^^ Use `ActiveSupport::Testing::TimeHelpers` instead of `Timecop`
+      RUBY
+    end
+  end
+
+  describe 'Timecop' do
+    it 'adds an offense' do
+      expect_offense(<<-RUBY.strip_indent)
+        Timecop.foo
```

Comment:
I think `.foo` case is covered in the example above already.

Reply:
I've removed the `.foo`, but I've left the example of just `Timecop` on its own, which I think is what I had meant this example to be in the first place.